include/dav1d/headers.rs

```diff
@@ -163,33 +166,68 @@ impl Dav1dWarpedMotionParams {
     }
 }
 
+#[derive(Clone, Copy, Default)]
+pub(crate) struct InnerAbcd([i16; 4]);
+
+impl Atom for InnerAbcd {
+    type Repr = u64;
+
+    fn pack(self) -> Self::Repr {
+        let [[b1, b2], [b3, b4], [b5, b6], [b7, b8]] = self.0.map(|x| x.to_ne_bytes());
+        Self::Repr::from_ne_bytes([b1, b2, b3, b4, b5, b6, b7, b8])
+    }
+
+    fn unpack(src: Self::Repr) -> Self {
+        let [b1, b2, b3, b4, b5, b6, b7, b8] = src.to_ne_bytes();
+        Self([[b1, b2], [b3, b4], [b5, b6], [b7, b8]].map(i16::from_ne_bytes))
+    }
+}
+
+#[derive(Default)]
+pub struct Abcd(Atomic<InnerAbcd>);
+
+impl Abcd {
+    pub fn new(abcd: [i16; 4]) -> Self {
+        Self(Atomic::new(InnerAbcd(abcd)))
+    }
+
+    pub fn get(&self) -> [i16; 4] {
+        self.0.load(Ordering::SeqCst).0
+    }
+
+    pub fn set(&self, abcd: [i16; 4]) {
+        self.0.store(InnerAbcd(abcd), Ordering::SeqCst);
+    }
+}
```

Comment:
Hmm, I think that should work, too.  I did want to add the `atomig` dependency in general, though, because there are other places where it'd be very useful, too, such as `#[derive(Atom)]` for data-less `enum`s.

Reply:
That sounds like a good case for adding `atomig` later, but I wouldn't use it in this case. Remove it as a dependency for now and re-add it once we need it for the enums.

Reply:
Sounds good.

Reply:
Fixed in da6ef27.